assets/js/src/core/modules/element/dynamic-types/definitions/field-filters/components/dynamic-type-field-filter-datetime-component.tsx

```diff
@@ -90,7 +91,19 @@ export const DynamicTypeFieldFilterDatetimeComponent = (props: DynamicTypeFieldF
     }
   }
 
-  const handleDateChange = (field: 'on' | 'from' | 'to', value: number | null): void => {
+  const convertValueToISOFormat = (timestamp: number | null): string | null => {
+    if (timestamp === null) return null
+
+    return dayjs.unix(timestamp).format(DATE_FORMAT)
```

Comment:
We should send a ISO date format with included timzone instead of just YYYY-MM-DD